### PR TITLE
Reqmgr2 MicroService deployment files

### PR DESCRIPTION
**Pull request by vkuznet**

### CODE REVIEW

File: reqmgr2_ms/deploy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    * )       base_url="https://`hostname -f`"          dbs_ins="private_vm";;
+  esac
+  perl -p -i -e "s{\"\@\@BASE_URL\@\@\"}{\"$base_url\"}g; \
+                 s{\"\@\@DBS_INS\@\@\"}{\"$dbs_ins\"}g"   \
+       $root/$cfgversion/config/$project/config.py
+}
+
+deploy_reqmgr2_post()
+{
+  case $host in vocms013[89] | vocms073[89] ) disable;;  * ) enable;; esac
```
Comment (amaltaro) [inline reqmgr2_ms/deploy]: We should update this list of hosts to reflect the same as workqueue, such that this service runs in a single node, initially at least.

File: reqmgr2_ms/manage [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+{
+  CHECK=$(echo "$1" | md5sum | awk '{print $1}')
+  if [ $CHECK != 94e261a5a70785552d34a65068819993 ]; then
+    echo "$0: cannot complete operation, please check documentation." 1>&2
+    exit 2;
+  fi
+}
+
+
+updateversions()
```
Comment (amaltaro) [inline reqmgr2_ms/manage]: I guess you took it from reqmgr2? This function is not needed

File: reqmgr2_ms/deploy (hunk 1 above)
Comment (amaltaro) [inline reqmgr2_ms/deploy]: We should update this list of hosts to reflect the same as workqueue, such that this service runs in a single node, initially at least.

**Comment by vkuznet** [comment]
Alan, thanks for prompt review, but I'm not yet finished. There are plenty of stuff to be clean-up. Once I'm done I'll send explicit request to review.

But, regarding your comment about single node. Which node should we use?

**Comment by amaltaro** [comment]
Ok, thanks Valentin. About the node, maybe we could start with the same node as global workqueue, so vocms0140 for production, vocms0131 for testbed.

File: frontend/app_reqmgr2ms_ssl.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+RewriteRule ^(/microservice(/.*)?)$ /auth/verify${escape:$1} [QSA,PT,E=AUTH_SPEC:cert]
+RewriteRule ^/auth/complete(/microservice(/.*)?)$ http://%{ENV:BACKEND}:8247${escape:$1} [QSA,P,L,NE]
```
Comment (ticoann) [inline frontend/app_reqmgr2ms_ssl.conf]: Is this port available?

File: reqmgr2ms/manage [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+##H   help              show this help
+##H   version           get current version of the service
+##H   status            show current service's status
+##H   sysboot           start server from crond if not running
+##H   restart           (re)start the service
+##H   start             (re)start the service
+##H   stop              stop the service
+##H
+##H For more details please refer to operations page:
+##H   https://twiki.cern.ch/twiki/bin/view/CMS/ReqMgrSystemDesign
```
Comment (ticoann) [inline reqmgr2ms/manage]: I guess we need use the MicroService doc if exists.

File: reqmgr2ms/manage [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+CFGFILE=$CFGDIR/ms_config.py
+AUTHDIR=$TOP/current/auth/reqmgr2
+COLOR_OK="\\033[0;32m"
+COLOR_WARN="\\033[0;31m"
+COLOR_NORMAL="\\033[0;39m"
+
+. $ROOT/apps/$ME/etc/profile.d/init.sh
+
+export PYTHONPATH=$ROOT/auth/$ME:$PYTHONPATH
+export WMCORE_ROOT=$REQMGR2_ROOT/lib/python*/site-packages
```
Comment (ticoann) [inline reqmgr2ms/manage]: shouldn't it be REQMGR2MS_ROOT?

File: reqmgr2ms/manage [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+if [ -e $AUTHDIR/dmwm-service-cert.pem ] && [ -e $AUTHDIR/dmwm-service-key.pem ]; then
+  export X509_USER_CERT=$AUTHDIR/dmwm-service-cert.pem
+  export X509_USER_KEY=$AUTHDIR/dmwm-service-key.pem
+fi
+
+
+# Start service conditionally on crond restart.
+sysboot()
+{
+  LD_PRELOAD=$JEMALLOC_ROOT/lib/libjemalloc.so wmc-httpd -v -d $STATEDIR -l "|rotatelogs $LOGDIR/reqmgr2-ms-%Y%m%d.log 86400" $CFGFILE
```
Comment (ticoann) [inline reqmgr2ms/manage]: reqmgr2ms-%Y%m%d.log - Just for the consistency?

File: reqmgr2ms/manage [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+
+# Start the service.
+start()
+{
+  echo "starting $ME"
+  LD_PRELOAD=$JEMALLOC_ROOT/lib/libjemalloc.so wmc-httpd -r -d $STATEDIR -l "|rotatelogs $LOGDIR/reqmgr2-%Y%m%d.log 86400" $CFGFILE
```
Comment (ticoann) [inline reqmgr2ms/manage]: reqmgr2ms-%Y%m%d.log - Just for the consistency?

File: reqmgr2ms/manage [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+{
+  wmc-httpd -s -d $STATEDIR $CFGFILE
+}
+
+
+# Verify the security string.
+check()
+{
+  CHECK=$(echo "$1" | md5sum | awk '{print $1}')
+  if [ $CHECK != 94e261a5a70785552d34a65068819993 ]; then
```
Comment (ticoann) [inline reqmgr2ms/manage]: Is this correct md5sum?

File: reqmgr2ms/manage [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    check "$2"
+    stop
+    ;;
+
+  help )
+    perl -ne '/^##H/ && do { s/^##H ?//; print }' < $0
+    ;;
+
+  version )
+    echo "$REQMGR2_VERSION"
```
Comment (ticoann) [inline reqmgr2ms/manage]: REQMGR2MC_VERSION

File: reqmgr2ms/monitoring.ini [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+# Perl regex to look for the service process using ps
+PS_REGEX="wmc-httpd.*[/]ms_config.py"
+
+# The ping test fetches the provided URL and look for the following perl regex
+PING_URL="http://localhost:8247/microservice/data/status"
+PING_HEADER="Accept: application/json"
+PING_REGEX="microservice"
+
+# Monitor process activity
+PROCESS_OWNER="_reqmgr2-ms"
```
Comment (ticoann) [inline reqmgr2ms/monitoring.ini]: _reqmgr2ms

File: frontend/app_reqmgr2ms_ssl.conf (hunk 3 above)
Comment (vkuznet) [reply frontend/app_reqmgr2ms_ssl.conf]: I took next port in 824X range which should be allocated to ReqMgr2 needs. But we'll check it with Lina.

File: reqmgr2ms/manage (hunk 4 above)
Comment (vkuznet) [reply reqmgr2ms/manage]: I rather prefer to add twiki page in cern domain and rather have reference to code itself in a wiki page of WMCore.

File: reqmgr2ms/manage (hunk 8 above)
Comment (vkuznet) [reply reqmgr2ms/manage]: yes, it corresponds to phrase operator use, something like I read documentation, instead of relates to specific service

**Pull request merged**